microsoft-edge/progressive-web-apps-chromium/how-to/app-actions.md

```diff
@@ -0,0 +1,377 @@
+         * [Web Manifest validation for launch_handler](#web-manifest-validation-for-launch_handler)
+         * [Accept entity info from Windows App Actions framework](#accept-entity-info-from-windows-app-actions-framework)
+   * [Package your PWA](#package-your-pwa)
+      * [How to package store PWA in Edge Stable](#how-to-package-store-pwa-in-edge-stable)
+      * [How to package store PWA in Edge Canary](#how-to-package-store-pwa-in-edge-canary)
+   * [Testing your PWA in Windows](#testing-your-pwa-in-windows)
+      * [Using Windows actions test tool](#using-windows-actions-test-tool)
+
+
+<!-- ====================================================================== -->
+## Introduction
+
+Here's how to author and test an app action for the App Actions on Windows framework, for a Progressive Web App (PWA) that you've created.
+
+
+<!-- ------------------------------ -->
+#### What's App Actions on Windows
+
+An App Action on Windows is an individual unit of behavior that a Windows app can implement and register.  The App Action can then be accessed from other apps and experiences, seamlessly integrating into user workflows. To learn more about App Actions on Windows, see [App Actions on Windows](https://learn.microsoft.com/en-us/windows/ai/app-actions/).
+
+An App Action is an atomic unit of functionality.  An app builds and registers actions, and then Windows or other apps can recommend registered actions to the user at contextually relevant times and locations within the user workflow.
+
+
+<!-- ------------------------------ -->
+#### Why should you build actions for your app
+
+App developers can build actions to increase discoverability and engagement for their features with app-to-app and agentic interactions.
+
+The Windows App Actions framework is for exposing and executing key pieces of app functionality – it is not for generic user activities like opening context menus or right-clicking.
+
+See [Recommended scenarios for App Actions](/windows/ai/app-actions#recommended-scenarios-for-app-actions) in _App Actions on Windows Overview_.
+
+
+<!-- ====================================================================== -->
+## Get started
+
+
+<!-- ------------------------------ -->
+#### Define the actions your PWA supports
+
+This guide illustrates how to author an action using one of the sample PWAs, [wami, the Web App to Manipulate Images](https://github.com/MicrosoftEdge/Demos/tree/main/wami).  The Wami demo supports image manipulation tasks, such as resizing or applying post-processing styles.  The following example demonstrates creating a **Paint** action, which generates an oil-painting-style version of the image.
+
+The **Paint** action accepts the image file that user selected and returns the stylized output.  The **Paint** action is defined in a descriptive file named `ActionsManifest.json`, per the next section.
+
+
+<!-- -------------------- -->
+##### Prepare ActionsManifest.json
```

Comment:
flatten toc eg remove this heading.
rendered:
https://github.com/mikehoffms/edge-developer/blob/user/mikehoffms/branch-02/microsoft-edge/progressive-web-apps-chromium/how-to/app-actions.md#prepare-actionsmanifestjson